HousingSearchApi/V1/Interfaces/SearchGateway.cs

```diff
@@ -58,9 +61,23 @@ public async Task<GetAssetListResponse> GetListOfAssets(HousingSearchRequest que
                 queryableAsset.Create())
             );
 
-            assetListResponse.SetTotal(searchResponse.Total);
+            assetListResponse.SetTotal(searchResponse.Total < 0 ? 0 : searchResponse.Total);
```

Comment:
the total could easily be worked out from within `assetListResponse` no and then you can move this logic inside that class?

Reply:
searchResponse.Total is read only!